src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -506,13 +506,16 @@
   <data name="BufferMaximumSizeExceeded" xml:space="preserve">
     <value>Cannot allocate a buffer of size {0}.</value>
   </data>
-  <data name="CannotSerializeOpenGeneric" xml:space="preserve">
-    <value>The type '{0}' has generic parameters that have not been replaced by specific types, which is not valid during serialization or deserialization.</value>
+  <data name="CannotSerializeInvalidType" xml:space="preserve">
+    <value>The type '{0}' contains generic parameters that have not been replaced by specific types, is a pointer type, or is a ref struct. This is invalid during serialization or deserialization.</value>
```

Comment:
```suggestion
    <value>The type '{0}' contains generic parameters that have not been replaced by specific types, are pointer types, or are ref structs. This is invalid during serialization or deserialization.</value>
```

Reply:
What I'm trying to express is that one of the following is true:

1. The type contains generic parameters that have not been replaced by specific types.
2. The type is a pointer.
3. The type is a ref struct.

Reply:
I understand.  I was fixing the grammar. "generic parameters...is" is incorrect... it should be "generic parameters...are".

Reply:
Maybe it should be `The type '{0}' contains a generic parameter that has not been replaced by specific types, is a pointer type, or is a ref struct type. This is invalid during serialization or deserialization.`.

Reply:
That's fine, too. Except "that has either" => "that either has".

Reply:
The subject here is "the type", i.e one is trying to (de)serialize a type that 

1. contains generic parameters that have not been replaced by specific types (e.g. the declared type is `typeof(List<>)`), or
2. is a pointer (e.g. the declared type is `typeof(int*)`), or
3. is a ref struct (e.g. the declared type is `typeof(Span<byte>)`).

Is the following message clearer?

```
The type '{0}' is not supported for serialization or deserialization, for one of the following reasons:

1. The type contains generic parameters that have not been replaced by specific types.
2. The type is a pointer.
3. The type is a ref struct.
```

Reply:
> The subject here is "the type", i.e one is trying to (de)serialize a type that

Ah, ok, that's not what it was saying, or at the very least it was ambiguous.

In that case, it'd be easiest to just reorder things:
```
The type '{0}' is invalid for serialization or deserialization because it is a pointer type, is a ref struct, or contains generic parameters that have not been replaced by specific types.
```

Reply:
+1 to @stephentoub suggestion